#==================================================================================================
# Upstream patch for dylib versioning. Expected to be included in next release.
#
# Author: Christopher Nielsen
#
# See: https://gitlab.freedesktop.org/pixman/pixman/-/issues/81
#==================================================================================================
--- meson.build.orig	2023-10-06 08:09:45.000000000 -0400
+++ meson.build	2023-10-06 08:17:58.000000000 -0400
@@ -578,10 +578,20 @@
 config.set('PACKAGE', 'foo')
 
 version_conf = configuration_data()
-split = meson.project_version().split('.')
-version_conf.set('PIXMAN_VERSION_MAJOR', split[0])
-version_conf.set('PIXMAN_VERSION_MINOR', split[1])
-version_conf.set('PIXMAN_VERSION_MICRO', split[2])
+ver_split = meson.project_version().split('.')
+proj_ver_major = ver_split[0]
+proj_ver_minor = ver_split[1]
+proj_ver_micro = ver_split[2]
+
+version_conf.set('PIXMAN_VERSION_MAJOR', proj_ver_major)
+version_conf.set('PIXMAN_VERSION_MINOR', proj_ver_minor)
+version_conf.set('PIXMAN_VERSION_MICRO', proj_ver_micro)
+
+proj_dylib_ver_compat = '39.0.0'
+proj_dylib_ver_current = '@0@.0.0'.format(proj_ver_minor)
+message('proj_dylib_ver_compat: ', proj_dylib_ver_compat)
+message('proj_dylib_ver_current: ', proj_dylib_ver_current)
+proj_darwin_versions = [proj_dylib_ver_compat, proj_dylib_ver_current]
 
 add_project_arguments('-DHAVE_CONFIG_H', language : ['c'])
 
--- pixman/meson.build.orig	2023-10-06 08:07:20.000000000 -0400
+++ pixman/meson.build	2023-10-06 08:11:52.000000000 -0400
@@ -125,6 +125,7 @@
   dependencies : [dep_m, dep_threads],
   include_directories : cpu_features_inc,
   version : meson.project_version(),
+  darwin_versions : proj_darwin_versions,
   install : true,
 )
 
